cmds/coredhcp/main.go

```diff
@@ -43,10 +62,14 @@ var desiredPlugins = []*plugins.Plugin{
 func main() {
 	flag.Parse()
 	log := logger.GetLogger("main")
-	if *flagDebug {
-		log.Logger.SetLevel(logrus.DebugLevel)
-		log.Infof("Enabled debug logging")
+	fn, ok := logLevels[*flagLogLevel]
+	if !ok {
+		log.Fatalf("Invalid log level '%s'. Valid log levels are %v", *flagLogLevel, getLogLevels())
+	}
+	if *flagLogLevel != "none" {
+		log.Infof("Setting log level to '%s'", *flagLogLevel)
 	}
+	fn(log.Logger)
```

Comment:
This should be first - if I pass loglevel critical, I don't want an info-level log to tell me I've set the loglevel to critical

Reply:
done

Reply:
I have also removed the "none" check which is not necessary anymore after the swap